optimum/intel/openvino/configuration.py

```diff
@@ -782,3 +848,92 @@ def to_dict(self) -> Dict[str, Any]:
+            (1) weights of weighted layers to the precision given in the `weight_quantization_config`, and
+            (2) weights and activations of other possible layers; precision is given in the `full_quantization_config`.
+
+        By default, weights of all weighted layers are quantized in the first step. In the second step activations of
+        weighted and non-weighted layers are quantized. If some layers are instructed to be ignored in the first step
+        with `weight_quantization_config.ignored_scope` parameter, both weights and activations of these layers are
+        quantized to the precision given in the `full_quantization_config`.
+
+        Args:
+            weight_quantization_config (`OVWeightQuantizationConfig` or `dict`):
+                Configuration related to weight quantization.
+            full_quantization_config (`OVQuantizationConfig` or `dict`):
+                Configuration related to full quantization.
+            ignored_scope (`dict` or `nncf.IgnoredScope`, *optional*):
+                An ignored scope that defines the list of model nodes to be ignored during quantization. Dictionary
+                entries provided via this argument are used to create an instance of `nncf.IgnoredScope` class.
+                Ignored scope provided here will be used for both weight and full quantization steps.
+            num_samples (`int`, *optional*):
+                The maximum number of samples composing the calibration dataset.
+            dataset (`str or List[str]`, *optional*):
+                The dataset used for data-aware optimization with NNCF.
+            tokenizer (`str`, *optional*):
+                The tokenizer used to process the dataset.
+            processor (`str`, *optional*):
+                A transformers processor used to process the dataset inputs.
+            trust_remote_code (`bool`, defaults to `False`):
+                Allows to use custom code for the modeling hosted in the model repository. This option should only be
+                set for repositories you trust and in which you have read the code, as it will execute on your local
+                machine arbitrary code present in the model repository.
+            **kwargs:
+        """
+        if isinstance(weight_quantization_config, dict):
+            weight_quantization_config = OVWeightQuantizationConfig.from_dict(weight_quantization_config)
+        else:
+            weight_quantization_config = weight_quantization_config.clone()
+        self.weight_quantization_config = weight_quantization_config
+        wqc = self.weight_quantization_config
+
+        if isinstance(full_quantization_config, dict):
+            full_quantization_config = OVQuantizationConfig.from_dict(full_quantization_config)
+        else:
+            full_quantization_config = full_quantization_config.clone()
+        self.full_quantization_config = full_quantization_config
+        fqc = self.full_quantization_config
```

Comment:
What do you think about using the `_initialize_quantization_config` method to handle the initialization of `weight_quantization_config` and `full_quantization_config` to avoid code duplication?